### ISSUE TITLE
Extract influxdb_line_protocol to independent crate

### ISSUE THREAD

**Issue by jiacai2050** [influxdata/influxdb_iox#7051]
__Use case:__

We at CeresDB would like to support write by influxdb line protocol, so I want to ask is there any possible to extract line_protocol module to an independent crate, and I would like help with this if maintainer agree with this change.

__Proposal:__

Extract line_protocol to an independent crate, and publish to crates.io.
__Current behaviour:__
Currently it's in monorepo of iox.

__Desired behaviour:__
Describe what you want.

__Alternatives considered:__
Describe other solutions or features you considered.

**Comment by jiacai2050** [comment]
@pauldix @alamb Would you mind share your ideas about this?

**Comment by alamb** [comment]
@jiacai2050  we are discussing this internally and will get back to you.

**Comment by alamb** [comment]
Hi @jiacai2050 -- we would be happy to help make the `influxdb_line_protocol` crate easier to use  for other projects. 

What would you think about initially keeping the code in the iox monorepo repo and publishing to [crates.io](http://crates.io/)? This would might keep the overhead (for the IOx team) lower as it would keep the number of repositories for us to manage down (and steps required to improve the line protocol parser)

If other people start using it or we want to build a bigger community around it, breaking it into its own repo would make sense.

**Comment by alamb** [comment]
In terms of next steps, here is the list I can think of:

- [ ] Review the current API and make sure it is reasonable for external use (kudos to @crepererum  for suggesting this). 
- [ ] Update Cargo.toml and write a readme, etc
- [ ] Publish to crates.io

@jiacai2050  do you have any thoughts / suggestions on API?

**Comment by jiacai2050** [comment]
Thanks for supports, really appreciated it.

> What would you think about initially keeping the code in the iox monorepo repo and publishing to [crates.io](http://crates.io/)?

That make sense to me.

> do you have any thoughts / suggestions on API?

I will review API more carefully today.

**Comment by jiacai2050** [comment]
The Rust API seems good for me.

Out of curiosity, I wonder when will FFI API will be used? Do you want to include FFI when publish?

**Comment by alamb** [comment]
> The Rust API seems good for me.

Thanks @jiacai2050  -- I will put this item on my list of things to work on over the next few weeks. 

> Out of curiosity, I wonder when will FFI API will be used? Do you want to include FFI when publish?

https://github.com/influxdata/influxdb_iox/blob/main/influxdb_line_protocol/Cargo.toml#L12

I think the FFI interface was added by @mkmik  in  #5267 -- I am not sure if it is currently used. @mkmik  is it still used do you know?

**Comment by mkmik** [comment]
It was used so I could compare the go and the rust implementations. It's currently no longer needed, and if I'll need it again I can easily add it again

**Comment by alamb** [comment]
Sounds good -- I will plan to to remove the FFI support as part of this project. Thank you @mkmik

**Comment by jiacai2050** [comment]
@alamb Thanks for your kind support, please ping me when crate is published.

**Comment by alamb** [comment]
I am beginning to work on this item actively

**Comment by alamb** [comment]
https://github.com/influxdata/influxdb_iox/pull/7195 is the first chunk, once that is merged I'll publish a `1.0.0` version to crates.io (we haven't really changed the API in several years)

**Comment by alamb** [comment]
@carols10cents  points out there is already a crate with the same name: https://crates.io/crates/influxdb_line_protocol though it looks like it has no activity for 2 years -- I left a note on https://github.com/Hakuyume/influxdb-rs/issues/7 to ask if we can have the crates.io entry.

**Comment by alamb** [comment]
@Hakuyume has graciously transferred ownership of the crates.io entry. We will continue towards release in a day or two.

**Comment by jiacai2050** [comment]
> @Hakuyume has graciously transferred ownership of the crates.io entry. We will continue towards release in a day or two.

👍

**Comment by alamb** [comment]
It is now available on crates.io https://crates.io/crates/influxdb-line-protocol (version 1.0.0!)

@jiacai2050  please feel free to give it a try!
